src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -23,6 +23,7 @@ case class NullConst extends ConstExpr
 case class FuncExpr(function: String, args: List[Expr], isAggregation: Boolean) extends Expr
 case class BinaryOpExpr(lhs: Expr, op: String, rhs: Expr) extends Expr
 case class TypecastExpr(lhs: Expr, rhs: String) extends Expr
+case class Placeholder extends Expr
```

Comment:
Thanks for the detailed explanation! A single variable has some special meaning compared to an expression, and we should treat them differently.